packages/graphql-codegen-typescript-operations/src/ts-selection-set-to-object.ts

```diff
@@ -0,0 +1,99 @@
+import { SelectionSetToObject as CodegenSelectionSetToObject } from "@graphql-codegen/visitor-plugin-common";
+import {
+  FieldNode,
+  FragmentSpreadNode,
+  GraphQLNamedType,
+  InlineFragmentNode,
+  Kind,
+  SelectionNode,
+  SelectionSetNode,
+} from "graphql";
+
+export class SelectionSetToObject extends CodegenSelectionSetToObject {
+  public createNext(
+    parentSchemaType: GraphQLNamedType,
+    selectionSet: SelectionSetNode,
+  ): SelectionSetToObject {
+    return new SelectionSetToObject(
+      this._processor,
+      this._scalars,
+      this._schema,
+      this._convertName.bind(this),
+      this._getFragmentSuffix.bind(this),
+      this._loadedFragments,
+      this._config,
+      parentSchemaType,
+      selectionSet,
+    );
+  }
+
+  protected flattenSelectionSet(
+    selections: ReadonlyArray<SelectionNode>,
+  ): Map<string, Array<SelectionNode | string>> {
+    const selectionNodesByTypeName = new Map<
+      string,
+      Array<SelectionNode | string>
+    >();
+    const inlineFragmentSelections: InlineFragmentNode[] = [];
+    const fieldNodes: FieldNode[] = [];
+    const fragmentSpreads: FragmentSpreadNode[] = [];
+
+    for (const selection of selections) {
+      if (selection.directives?.some((value) => value.name.value === "mask")) {
+        break;
+      }
```

Comment:
Mask should be only applicable to FRAGMENT_SPREAD. When seen on other selection kinds (FIELD, INLINE_FRAGMENT) we should throw instead.